source/common/ssl/ssl_socket.h

```diff
@@ -55,6 +56,7 @@ class SslSocket : public Network::TransportSocket,
   std::vector<std::string> getDnsSansFromCertificate(X509* cert);
 
   Network::TransportSocketCallbacks* callbacks_{};
+  ContextSharedPtr ctx_owner_;
```

Comment:
you should be able to have this as `std::shared_ptr<ContextImpl>` and cast it with `std::dynamic_pointer_cast`

Reply:
As I mentioned above,  I try to remove this member variable ctx_.